packages/dds/tree/src/simple-tree/core/allowedTypes.ts

```diff
@@ -131,7 +131,30 @@ export interface AllowedTypeMetadata {
 	 */
 	readonly custom?: unknown;
 
-	// TODO metadata for enablable types will be added here
+	/**
+	 * If defined, indicates that an allowed type is staged. Before upgrade, any attempt to insert or move a node to the location where this allowed type is declared will throw an error.
+	 */
+	readonly stagedSchemaUpgrade?: SchemaUpgrade;
+}
+
+/**
+ * Package internal construction API.
+ */
+export let createSchemaUpgrade: () => SchemaUpgrade;
+
+/**
+ * Unique token used to upgrade schemas and determine if a particular upgrade has been completed.
+ *
+ * TODO:#38722 implement runtime schema upgrades until then, the class purely behaves as a placeholder and we disable no-extraneous-class
+ * @sealed @alpha
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class SchemaUpgrade {
```

Comment:
Why is this a class? Why not a symbol? And later on, simply a bool/function, etc?

Reply:
Using classes like this is the cleanest way in typescript to be able to make strongly typed objects which users cannot construct.

Each schema upgrade gets its own unique identity.

I'm not sure how you would use symbols Booleans or functions to define a type which can hold values with distinct identities that are not user constructable.